Xamarin.PropertyEditing.Tests/MockControls/MockResourceProvider.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xamarin.PropertyEditing.Drawing;
+
+namespace Xamarin.PropertyEditing.Tests
+{
+	public class MockResourceProvider
+		: IResourceProvider
+	{
+		public Task<Resource> CreateResourceAsync<T> (ResourceSource source, string name, T value)
+		{
+			throw new NotImplementedException ();
+		}
+
+		public Task<IReadOnlyList<Resource>> GetResourcesAsync (object target, IPropertyInfo property, CancellationToken cancelToken)
+		{
+			return Task.FromResult<IReadOnlyList<Resource>> (SystemResources.Where (r => property.Type.IsAssignableFrom (r.GetType().GetGenericArguments()[0])).ToList());
+		}
+
+		public Task<IReadOnlyList<ResourceSource>> GetResourceSourcesAsync (object target, IPropertyInfo property)
+		{
+			return Task.FromResult<IReadOnlyList<ResourceSource>> (new[] { SystemResourcesSource });
+		}
+
+		internal static readonly ResourceSource SystemResourcesSource = new ResourceSource ("System Resources", isLocal: false);
```

Comment:
We shouldn't hard-code the actual resources. The client code, whether it's test case or the standalone apps, should set those.

Reply:
Test cases should largely be using inline `Mock<>`, not this. The standalone apps I'm not really worried about being that flexible, we can adapt it later if need be.